metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -51,31 +51,57 @@ case class SbtBuildTool(version: String) extends BuildTool {
           sbtArgs
         ).flatten
     }
-    writeSbtMetalsPlugin(config)
+    removeLegacyGlobalPlugin()
+    writeSbtMetalsPlugin(workspace, config)
+    gitignoreMetals(workspace)
     allArgs
   }
+
   override def digest(
       workspace: AbsolutePath
   ): Option[String] = SbtDigest.current(workspace)
   override val minimumVersion: String = "0.13.17"
 
-  private def writeSbtMetalsPlugin(config: MetalsServerConfig): Unit = {
+  // We remove legacy metals.sbt file that was located in
+  // global sbt plugins and which adds the plugin to each projects
+  // and creates additional overhead.
+  private def removeLegacyGlobalPlugin(): Unit = {
+    def pluginsDirectory(version: String): AbsolutePath = {
+      AbsolutePath(System.getProperty("user.home"))
+        .resolve(".sbt")
+        .resolve(version)
+        .resolve("plugins")
+    }
     val plugins =
-      if (version.startsWith("0.13")) SbtBuildTool.pluginsDirectory("0.13")
-      else SbtBuildTool.pluginsDirectory("1.0")
-    plugins.createDirectories()
+      if (version.startsWith("0.13")) pluginsDirectory("0.13")
+      else pluginsDirectory("1.0")
+
+    val metalsFile = plugins.resolve("metals.sbt")
+    Files.deleteIfExists(metalsFile.toNIO)
+  }
+
+  private def gitignoreMetals(workspace: AbsolutePath) = {
+    val gitignore = workspace.resolve(".gitignore")
+    val gitIgnoreContents = "project/metals.sbt"
```

Comment:
What do you think about adding the following entries as well?
```
.metals/
.bloop/
```
These entries should ideally be in the workspace .gitignore instead of the global .gitignore because VS Code only uses the workspace gitignore in "Open file".

Reply:
I was thinking about it, but decided not to do in this PR, but I can for sure do it here.